app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -44,6 +45,15 @@ class LuciBuildService {
     this.log = log;
   }
 
+  /// Shards [rows] into several sublists of size [maxEntityGroups].
+  Future<List<List<Request>>> shard(List<Request> requests, int max) async {
```

Comment:
Curious what caused the index issue you hit? Some tests on this function should help.

Reply:
it was a problem with trying to schedule a build that didn't exist. Turns out that modifying presubmit tags requires to also provide the bringup property.